src/coreclr/vm/readytoruninfo.h

```diff
@@ -48,6 +48,42 @@ class ReadyToRunCoreInfo
 typedef DPTR(class ReadyToRunInfo) PTR_ReadyToRunInfo;
 typedef DPTR(class ReadyToRunCoreInfo) PTR_ReadyToRunCoreInfo;
 
+
+class ReadyToRun_EnclosingTypeMap
+{
+private:
+    uint16_t TypeCount = 0;
+public:
+    static ReadyToRun_EnclosingTypeMap EmptyInstance;
+
+    mdTypeDef GetEnclosingType(mdTypeDef input, IMDInternalImport* pImport);
+    HRESULT GetEnclosingTypeNoThrow(mdTypeDef input, mdTypeDef *pEnclosingType, IMDInternalImport* pImport);
+};
+
+class ReadyToRun_TypeGenericInfoMap
+{
+private:
+    uint32_t TypeCount = 0;
+    ReadyToRunTypeGenericInfo GetTypeGenericInfo(mdTypeDef input, bool *foundResult);
+public:
+    static ReadyToRun_TypeGenericInfoMap EmptyInstance;
+
+    HRESULT IsGenericNoThrow(mdTypeDef input, bool *pIsGeneric, IMDInternalImport* pImport);
+    HRESULT GetGenericArgumentCountNoThrow(mdTypeDef input, uint32_t *pCount, IMDInternalImport* pImport);
+    bool IsGeneric(mdTypeDef input, IMDInternalImport* pImport);
+    uint32_t GetGenericArgumentCount(mdTypeDef input, IMDInternalImport* pImport);
+    bool HasVariance(mdTypeDef input, bool *foundResult);
+    bool HasConstraints(mdTypeDef input, bool *foundResult);
+};
+
+class ReadyToRun_MethodIsGenericMap
+{
+    uint32_t MethodCount = 0;
+public:
+    static ReadyToRun_MethodIsGenericMap EmptyInstance;
```

Comment:
Can all these `EmptyInstance` values be marked `const` or is that going to mess with things?